v0/destinations/appcues/transform.js

```diff
@@ -0,0 +1,184 @@
+const get = require("get-value");
+const set = require("set-value");
+const {
+  EventType
+} = require("../../../constants");
+
+const {
+  defaultRequestConfig,
+  getFieldValueFromMessage,
+  getValueFromMessage,
+  removeUndefinedAndNullValues,
+  removeUndefinedAndNullAndEmptyValues,
+} = require("../../util");
+
+const {
+  ConfigCategory,
+  mappingConfig,
+  getEndpoint,
+} = require("./config");
+
+const {
+  utcTimeToEpoch,
+} = require("./utils");
+
+function buildResponse(message, payload, endpoint) {
+  const response = defaultRequestConfig();
+  response.endpoint = endpoint;
+  response.userId = message.userId || message.anonymousId;
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+  return {
+    ...response,
+    headers: {
+      "Content-Type": "application/json",
+    }
+  };
+}
+
+function getIdentifyPayload(message) {
+  let payload = {};
+  if (message.messageId) {
+    payload.request_id = message.messageId;
+  }
+  const traits = getFieldValueFromMessage(message, "traits");
+  if (traits && Object.keys(traits).length > 0) {
+    payload.profile_update = traits;
+  }
+  return payload;
+}
+
+function processIdentify(message, destination) {
+  return buildResponse(
+    message,
+    getIdentifyPayload(message),
+    getEndpoint(destination.Config.accountId, message.userId)
+  );
+}
+
+function getTrackPayload(message, mappingJson) {
+  let eventJson = {};
+  
+  // iterate over the destKeys and set the value if present
+  Object.keys(mappingJson).forEach(destKey => {
+    let value = get(message, mappingJson[destKey]);
+    if (value) {
+      eventJson[destKey] = value;
+    }
+  });
```

Comment:
Let's move it to the configJSON format we already support through `constructPayload`. Else we're losing the functionality of `required` checking and rewriting almost the same code here.

Reply:
Yes,  moved it